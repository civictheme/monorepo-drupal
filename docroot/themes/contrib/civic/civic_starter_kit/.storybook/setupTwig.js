const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
