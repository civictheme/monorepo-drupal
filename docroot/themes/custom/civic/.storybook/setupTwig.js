const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
