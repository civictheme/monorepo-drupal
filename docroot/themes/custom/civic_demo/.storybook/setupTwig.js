const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
