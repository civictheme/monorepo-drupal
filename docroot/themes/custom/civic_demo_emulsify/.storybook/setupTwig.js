const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
