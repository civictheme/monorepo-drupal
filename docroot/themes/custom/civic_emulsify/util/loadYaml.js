import { resolve } from 'path';
import { readFileSync } from 'fs';
import { parse } from 'yaml';
import R from 'ramda';

/**
 * Small utility function that loads a yaml file and parses it synchronously.
 * This is intended to make composition cleaner.
 *
 * @param {string} relativePath - relative path to a yaml file that will be loaded and parsed.
 *
 * @returns {string} JavaScript object that results from the yaml parsing of the specified file.
 */
export default function loadYaml(relativePath) {
  const fullPath = resolve(__dirname, relativePath);
  return parse(readFileSync(fullPath, 'utf8'));
}
