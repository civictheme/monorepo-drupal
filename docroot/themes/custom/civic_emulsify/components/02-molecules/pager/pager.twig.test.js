import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import loadYaml from '../../../util/loadYaml';
import { setupTwig } from '../../../.storybook/setupTwig';

setupTwig(Twig);

describe('pager', () => {
  it('can render a pager', async () => {
    const { container } = await render(
      join(__dirname, 'pager.twig'),
      loadYaml(join(__dirname, 'pager.yml')),
    );

    expect(container).toMatchSnapshot();
  });

  it('can render a pager with ellipses', async () => {
    const { container } = await render(
      join(__dirname, 'pager.twig'),
      loadYaml(join(__dirname, 'pager-ellipses.yml')),
    );

    expect(container).toMatchSnapshot();
  });

  it('can render a pager with prev ellipses', async () => {
    const { container } = await render(
      join(__dirname, 'pager.twig'),
      loadYaml(join(__dirname, 'pager-prev-ellipses.yml')),
    );

    expect(container).toMatchSnapshot();
  });

  it('can render a pager with prev and next ellipses', async () => {
    const { container } = await render(
      join(__dirname, 'pager.twig'),
      loadYaml(join(__dirname, 'pager-both-ellipses.yml')),
    );

    expect(container).toMatchSnapshot();
  });
});
