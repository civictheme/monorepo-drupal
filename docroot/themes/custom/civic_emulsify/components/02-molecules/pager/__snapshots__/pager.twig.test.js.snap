// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pager can render a pager 1`] = `
<div>
  
  
  <nav
    aria-labelledby=""
    class="pager"
    role="navigation"
  >
    
    
    <h4
      class="visually-hidden"
      id=""
    >
      Pagination
    </h4>
    
    
    <ul
      class="pager__items js-pager__items"
    >
      
      
            
            
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          0
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item  is-active"
      >
        
                    
        <a
          class="pager__link  is-active"
          href="#"
          title="Current page"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Current page
            
          </span>
          1
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          2
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          3
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          4
        </a>
        
        
      </li>
      
            
            
          
    </ul>
    
  
  </nav>
  

</div>
`;

exports[`pager can render a pager with ellipses 1`] = `
<div>
  
  
  <nav
    aria-labelledby=""
    class="pager"
    role="navigation"
  >
    
    
    <h4
      class="visually-hidden"
      id=""
    >
      Pagination
    </h4>
    
    
    <ul
      class="pager__items js-pager__items"
    >
      
      
            
            
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          0
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          1
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          2
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          3
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          4
        </a>
        
        
      </li>
      
            
              
      <li
        class="pager__item pager__item--ellipsis"
        role="presentation"
      >
        …
      </li>
      
            
              
      <li
        class="pager__item pager__item--next"
      >
        
          
        <a
          class="pager__link pager__link--next"
          href="#"
          rel="next"
          title="Go to next page"
        >
          
            
          <span
            class="visually-hidden"
          >
            Next page
          </span>
          
            
          <span
            aria-hidden="true"
          >
            Next ›
          </span>
          
          
        </a>
        
        
      </li>
      
          
    </ul>
    
  
  </nav>
  

</div>
`;

exports[`pager can render a pager with prev and next ellipses 1`] = `
<div>
  
  
  <nav
    aria-labelledby=""
    class="pager"
    role="navigation"
  >
    
    
    <h4
      class="visually-hidden"
      id=""
    >
      Pagination
    </h4>
    
    
    <ul
      class="pager__items js-pager__items"
    >
      
      
              
      <li
        class="pager__item pager__item--prev"
      >
        
          
        <a
          class="pager__link pager__link--prev"
          href="#"
          rel="prev"
          title="Go to previous page"
        >
          
            
          <span
            class="visually-hidden"
          >
            Previous page
          </span>
          
            
          <span
            aria-hidden="true"
          >
            ‹ Previous
          </span>
          
          
        </a>
        
        
      </li>
      
            
              
      <li
        class="pager__item pager__item--ellipsis"
        role="presentation"
      >
        …
      </li>
      
            
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          0
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          1
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          2
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          3
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          4
        </a>
        
        
      </li>
      
            
              
      <li
        class="pager__item pager__item--ellipsis"
        role="presentation"
      >
        …
      </li>
      
            
              
      <li
        class="pager__item pager__item--next"
      >
        
          
        <a
          class="pager__link pager__link--next"
          href="#"
          rel="next"
          title="Go to next page"
        >
          
            
          <span
            class="visually-hidden"
          >
            Next page
          </span>
          
            
          <span
            aria-hidden="true"
          >
            Next ›
          </span>
          
          
        </a>
        
        
      </li>
      
          
    </ul>
    
  
  </nav>
  

</div>
`;

exports[`pager can render a pager with prev ellipses 1`] = `
<div>
  
  
  <nav
    aria-labelledby=""
    class="pager"
    role="navigation"
  >
    
    
    <h4
      class="visually-hidden"
      id=""
    >
      Pagination
    </h4>
    
    
    <ul
      class="pager__items js-pager__items"
    >
      
      
              
      <li
        class="pager__item pager__item--prev"
      >
        
          
        <a
          class="pager__link pager__link--prev"
          href="#"
          rel="prev"
          title="Go to previous page"
        >
          
            
          <span
            class="visually-hidden"
          >
            Previous page
          </span>
          
            
          <span
            aria-hidden="true"
          >
            ‹ Previous
          </span>
          
          
        </a>
        
        
      </li>
      
            
              
      <li
        class="pager__item pager__item--ellipsis"
        role="presentation"
      >
        …
      </li>
      
            
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          0
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          1
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          2
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          3
        </a>
        
        
      </li>
      
                      
      <li
        class="pager__item "
      >
        
                    
        <a
          class="pager__link "
          href="#"
          title="Go to page @key"
        >
          
            
          <span
            class="visually-hidden"
          >
            
              Page 
            
          </span>
          4
        </a>
        
        
      </li>
      
            
            
          
    </ul>
    
  
  </nav>
  

</div>
`;
